### PR TITLE
feat: Llama Guard 3 Cookbook

### PR DESCRIPTION
**Pull request by dloman118**

### CODE REVIEW

File: llama-guard-safe-chatbot/Llama-Guard-3-Ensuring-Safe-Chatbot.ipynb [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    "\n",
+    "[Meta's Llama Guard 3](https://llama.meta.com/docs/model-cards-and-prompt-formats/llama-guard-3/) is an advanced content moderation tool designed to assess user messages for harmful content. By analyzing incoming messages, Llama Guard 3 can effectively identify and classify harmful content across 14 categories, including hate speech, threats, and misinformation. This proactive approach allows chatbots to filter out harmful messages, ensuring that the language model (LLM) does not engage with or respond to inappropriate content.\n",
+    "\n",
+    "In this tutorial, we will demonstrate how utilizing Llama Guard 3 on Groq API effectively detects harmful content in user messages. By detecting harmful messages from users, you can ensure that your chatbot either does not respond or provides a generic or safe response that does not engage with the harmful content. We will provide examples of how the model identifies and classifies harmful messages across various categories, ensuring that inappropriate content is filtered out before the chatbot engages with users.\n",
```
Comment (hozen-groq) [inline llama-guard-safe-chatbot/Llama-Guard-3-Ensuring-Safe-Chatbot.ipynb:20]:
```suggestion
    "In this tutorial, we will build a chatbot to demonstrate how utilizing Llama Guard 3 with Groq API effectively detects harmful content in user messages. By detecting harmful messages from users, you can ensure that your chatbot either does not respond or provides a generic or safe response that does not engage with the harmful content. We will provide examples of how the model identifies and classifies harmful messages across various categories, ensuring that inappropriate content is filtered out before the chatbot engages with users.\n",
```

File: llama-guard-safe-chatbot/Llama-Guard-3-Ensuring-Safe-Chatbot.ipynb [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+   "execution_count": 1,
+   "id": "55774e3d-d352-4226-ac1b-df581f7452aa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from groq import Groq\n",
+    "import pandas as pd\n",
+    "import random\n",
+    "\n",
+    "client = Groq()"
```
Comment (hozen-groq) [inline llama-guard-safe-chatbot/Llama-Guard-3-Ensuring-Safe-Chatbot.ipynb:44]: are you going to pass in the API key as an env variable here and mention in the notebook or just leave it like this?

File: llama-guard-safe-chatbot/Llama-Guard-3-Ensuring-Safe-Chatbot.ipynb [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+    "\n",
+    "print(chat_completion.choices[0].message.content)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "133d15c8-9a8a-4cae-a07e-998384c1d8eb",
+   "metadata": {},
+   "source": [
+    "We've provided Llama Guard 3 a user message, and it successfully identified the content as unsafe and in the S13 taxonomy (Elections). Now, let's try Llama Guard 3 with several prompts to see how effectively it classfies them. We'll use `llama-3.1-8b-instant` to generate example user messages, randomly choosing safe or unsafe, and run them through Llama Guard 3 to determine it."
```
Comment (hozen-groq) [inline llama-guard-safe-chatbot/Llama-Guard-3-Ensuring-Safe-Chatbot.ipynb:185]:
```suggestion
    "We've provided Llama Guard 3 a user message, and it successfully identified the content as unsafe and in the S13 taxonomy (Elections). Now, let's try Llama Guard 3 with several prompts to see how effectively it classifies them. We'll use `llama-3.1-8b-instant` to generate example user messages, randomly choosing safe or unsafe, and run them through Llama Guard 3 for classification"
```

File: llama-guard-safe-chatbot/Llama-Guard-3-Ensuring-Safe-Chatbot.ipynb (hunk 2 above)
Comment (dloman118) [reply llama-guard-safe-chatbot/Llama-Guard-3-Ensuring-Safe-Chatbot.ipynb:44]: Yeah good point, will be clearer to pass in the key explicitly with the env variable

**Pull request merged**